Sloth.Core/Jobs/AggieEnterpriseJournalJob.cs

```diff
@@ -0,0 +1,118 @@
+                {
+                    log.Information("No scheduled transactions found");
+                    return;
+                }
+
+                // group transactions by origin code and feed
+                var groups = transactions.GroupBy(t => t.Source);
+                foreach (var group in groups)
+                {
+                    try
+                    {
+                        var source = group.Key;
+                        var groupedTransactions = group.ToList();
+
+                        var originCode = source.OriginCode;
+
+                        log.Information("Processing {TransactionCount} transactions for {SourceName}",
+                            groupedTransactions.Count, source.Name);
+
+                        // loop through grouped transactions and upload to Aggie Enterprise, saving status of each
+                        foreach (var transaction in groupedTransactions)
+                        {
+                            try
+                            {
+                                var result = await _aggieEnterpriseService.CreateJournal(source, transaction);
+                                var requestStatus = result.GlJournalRequest.RequestStatus;
+
+                                // here we will store the result of the transaction upload
+                                var journalRequest = new JournalRequest
+                                    { Transactions = new[] { transaction }, Source = source };
+
+                                if (requestStatus.RequestId.HasValue &&
+                                    requestStatus.RequestStatus == RequestStatus.Pending)
+                                {
+                                    // success, update transaction status to uploaded
+                                    transaction.SetStatus(TransactionStatuses.Processing);
+
+                                    journalRequest.RequestId = requestStatus.RequestId.Value;
+                                    journalRequest.Status = requestStatus.RequestStatus.ToString();
+
+                                    // save journal request
+                                    _context.JournalRequests.Add(journalRequest);
+                                }
+                                else if (requestStatus.RequestId.HasValue &&
+                                         requestStatus.RequestStatus == RequestStatus.Rejected)
+                                {
+                                    // failure, update transaction status to rejected
+                                    transaction.SetStatus(TransactionStatuses.Rejected);
+
+                                    journalRequest.RequestId = requestStatus.RequestId.Value;
+                                    journalRequest.Status = requestStatus.RequestStatus.ToString();
+                                }
+
+                                // TODO: These are likely the only two statuses possible for a new request, but confirm
```

Comment:
https://ucdavis-adminit.github.io/erp-api-docs/api/index.html#definition-RequestStatus

Reply:
Agreed.... I had the same thought and for now I figured "keep trying" is what we want during testing